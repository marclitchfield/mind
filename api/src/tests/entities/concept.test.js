import { applySetup, verifyMutations } from '../test-setup';
import gql from 'graphql-tag';

describe('Concept', () => {
  applySetup();

  test('createInSpace', () => {
    return verifyMutations('Concept', 'createInSpace', {
      mutations: [
        ({space}) => gql`mutation { 
          Concept { concept:createInSpace(spaceId:"${space.id}" input:{title: "Test"}) { id } }
        }`
      ],
      query: ({space}) => gql`query { 
        Space(id:"${space.id}") { 
          concepts { title spaces { title } }
          rootConcepts { title spaces { title } }
        } 
      }`
    });
  });

  test('createSubConcept', () => {
    return verifyMutations('Concept', 'createSubConcept', {
      mutations: [
        ({space})  => gql`mutation { Concept { sup:createInSpace(spaceId:"${space.id}" input:{title: "Test"}) { id } } }`,
        ({sup}) => gql`mutation { Concept { sub:createSubConcept(superConceptId:"${sup.id}" input:{title: "Test"}) { id } } }`,
      ],
      query: ({sup}) => gql`query { 
        Concept(id:"${sup.id}") { 
          subConcepts { title superConcepts { title } }
        }
      }`
    });
  });

  test('addSubConcept', () => {
    return verifyMutations('Concept', 'addSubConcept', {
      mutations: [
        ({space}) => gql`mutation { Concept { sup:createInSpace(spaceId:"${space.id}" input:{title: "Test"}) { id } } }`,
        ({space}) => gql`mutation { Concept { sub:createInSpace(spaceId:"${space.id}" input:{title: "Test"}) { id } } }`,
        ({sup, sub}) => gql`mutation { Concept { addSubConcept(id:"${sup.id}" subConceptId:"${sub.id}") } }`,
      ],
      query: ({sup}) => gql`query { 
        Concept(id:"${sup.id}") { 
          subConcepts { title superConcepts { title } }
        }
      }`
    });
  }); 

  test('removeSubConcept', () => {
    return verifyMutations('Concept', 'addSubConcept', {
      mutations: [
        ({space}) => gql`mutation { Concept { sup:createInSpace(spaceId:"${space.id}" input:{title: "Test"}) { id } } }`,
        ({space}) => gql`mutation { Concept { sub:createInSpace(spaceId:"${space.id}" input:{title: "Test"}) { id } } }`,
        ({sup, sub}) => gql`mutation { Concept { addSubConcept(id:"${sup.id}" subConceptId:"${sub.id}") } }`,
        ({sup, sub}) => gql`mutation { Concept { removeSubConcept(id:"${sup.id}" subConceptId:"${sub.id}") } }`,
      ],
      query: ({sup}) => gql`query { 
        Concept(id:"${sup.id}") { 
          subConcepts { title superConcepts { title } }
        }
      }`
    });
  }); 
});