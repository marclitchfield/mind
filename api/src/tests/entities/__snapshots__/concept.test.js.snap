// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Concept addSubConcept 1`] = `
Object {
  "Concept": Object {
    "__typename": "Concept",
    "subConcepts": Array [
      Object {
        "__typename": "Concept",
        "superConcepts": Array [
          Object {
            "__typename": "Concept",
            "title": "Test",
          },
        ],
        "title": "Test",
      },
    ],
  },
}
`;

exports[`Concept createInSpace 1`] = `
Object {
  "Space": Object {
    "__typename": "Space",
    "concepts": Array [
      Object {
        "__typename": "Concept",
        "spaces": Array [
          Object {
            "__typename": "Space",
            "title": "createInSpace",
          },
        ],
        "title": "Test",
      },
    ],
    "rootConcepts": Array [
      Object {
        "__typename": "Concept",
        "spaces": Array [
          Object {
            "__typename": "Space",
            "title": "createInSpace",
          },
        ],
        "title": "Test",
      },
    ],
  },
}
`;

exports[`Concept createSubConcept 1`] = `
Object {
  "Concept": Object {
    "__typename": "Concept",
    "subConcepts": Array [
      Object {
        "__typename": "Concept",
        "superConcepts": Array [
          Object {
            "__typename": "Concept",
            "title": "Test",
          },
        ],
        "title": "Test",
      },
    ],
  },
}
`;
